static/app/views/app/index.tsx

```diff
@@ -35,201 +31,151 @@ const GlobalNotifications = HookOrDefault({
-    this.setState({
-      newsletterConsentPrompt: false,
-    });
+  function clearUpgrade() {
+    ConfigStore.set('needsUpgrade', false);
+  }
 
-  handleGlobalModalClose = () => {
-    if (typeof this.mainContainerRef.current?.focus === 'function') {
-      // Focus the main container to get hotkeys to keep working after modal closes
-      this.mainContainerRef.current.focus();
-    }
-  };
+  function clearNewsletterConsent() {
+    const flags = {...config.user.flags, newsletter_consent_prompt: false};
+    ConfigStore.set('user', {...config.user, flags});
+  }
 
-  renderBody() {
-    const {needsUpgrade, newsletterConsentPrompt} = this.state;
+  const needsUpgrade = config.user?.isSuperuser && config.needsUpgrade;
+  const newsletterConsentPrompt = config.user?.flags?.newsletter_consent_prompt;
 
+  function renderBody() {
     if (needsUpgrade) {
       const InstallWizard = lazy(() => import('app/views/admin/installWizard'));
 
       return (
         <Suspense fallback={null}>
-          <InstallWizard onConfigured={this.onConfigured} />;
+          <InstallWizard onConfigured={clearUpgrade} />;
         </Suspense>
       );
     }
 
     if (newsletterConsentPrompt) {
-      return <NewsletterConsent onSubmitSuccess={this.handleNewsletterConsent} />;
-    }
-
-    return this.props.children;
-  }
+      const NewsletterConsent = lazy(() => import('app/views/newsletterConsent'));
 
-  render() {
-    if (this.state.loading) {
       return (
-        <LoadingIndicator triangle>
-          {t('Getting a list of all of your organizations.')}
-        </LoadingIndicator>
+        <Suspense fallback={null}>
+          <NewsletterConsent onSubmitSuccess={clearNewsletterConsent} />
+        </Suspense>
       );
     }
 
-    return (
-      <MainContainer tabIndex={-1} ref={this.mainContainerRef}>
-        <GlobalModal onClose={this.handleGlobalModalClose} />
-        <SystemAlerts className="messages-container" />
-        <GlobalNotifications
-          className="notifications-container messages-container"
-          organization={this.state.organization}
-        />
-        <Indicators className="indicators-container" />
-        <ErrorBoundary>{this.renderBody()}</ErrorBoundary>
-      </MainContainer>
-    );
+    return children;
   }
+
+  // Used to restore focus to the container after closing the modal
+  const mainContainerRef = useRef<HTMLDivElement>(null);
+
+  return (
+    <MainContainer tabIndex={-1} ref={mainContainerRef}>
+      <GlobalModal onClose={() => mainContainerRef.current?.focus?.()} />
+      <SystemAlerts className="messages-container" />
+      <GlobalNotifications
+        className="notifications-container messages-container"
+        organization={organization ?? undefined}
+      />
+      <Indicators className="indicators-container" />
+      <ErrorBoundary>{renderBody()}</ErrorBoundary>
+    </MainContainer>
+  );
 }
 
-export default withApi(withConfig(App));
+export default withConfig(App);
```

Comment:
Would be nice if the `useLegacyStore` could work for ConfigStore, but unfortunately it does not because it doesn't match the interface.